### PR TITLE
MISUV-7394 Dymanic data for poa amount

### PR DESCRIPTION
**Pull request by danielmaltz1**

### CODE REVIEW

**Comment by simondrugan16** [comment]
Maybe some of this could be moved out of the controller?

File: app/controllers/SubmitPoaController.scala [hunk 1]
```diff
@@ -61,7 +70,147 @@ class SubmitPoaController @Inject()(cc: MessagesControllerComponents,
     }
   }
 
-  def extractNino(json: JsValue): Option[String] = {
+  private def overwriteTotalAmount(nino: String, json: JsValue): Unit = {
+    (extractPoAAmount(json), extractTaxYear(json)) match {
```
Comment (avelx) [inline app/controllers/SubmitPoaController.scala:74]: if we can refactor this method pls? you can move nested pattern matching into private methods etc

File: app/controllers/SubmitPoaController.scala [hunk 2]
```diff
@@ -61,7 +70,147 @@ class SubmitPoaController @Inject()(cc: MessagesControllerComponents,
...
+      case (Some(amount), Some(taxYearString)) => {
+        TaxYear.createTaxYearGivenTaxYearRange(taxYearString) match {
+          case Some(taxYear) => {
+            val financialUrl = getFinancialDetailsUrl(nino, taxYear)
+            val financialDetailsResponse = dataRepository.find(equal("_id", financialUrl))
+            financialDetailsResponse.map {
+              case Some(value) => value.response match {
+                case Some(response) =>
+                  //extract chargeReference for 1554 data changes
+                  val chargeRef = extractChargeRef(response).getOrElse("None")
```
Comment (avelx) [inline app/controllers/SubmitPoaController.scala:84]: I think you mean ?
`val chargeRef = extractChargeRef(response).getOrElse(None) `

File: app/controllers/SubmitPoaController.scala [hunk 3]
```diff
@@ -61,7 +70,147 @@ class SubmitPoaController @Inject()(cc: MessagesControllerComponents,
...
+                  //Create new 1553 data with totalAmount overwritten with new poa amount
+                  val newResponse = response.transform(transformDocDetails(amount.toInt)).getOrElse(response)
+                  //IF chargeRef exists, just update the totalAmount
+                  //Otherwise, add a chargeRef to 1553, and create 1554 data
+                  if (chargeRef == "None") {
```
Comment (avelx) [inline app/controllers/SubmitPoaController.scala:89]: if (chargeRef.isDefined)

**Comment by danielmaltz1** [comment]
Do not merge, changes needed

**Pull request merged**